### PR TITLE
feat: EsDataTableSimple molecule for es-ds 3.0

### PR DESCRIPTION
**Pull request by jaissica**
### 🔗 Linked issue

- https://energysage.atlassian.net/browse/CED-1743

### ❓ Type of change

- [ ] 📖 Documentation (updates to the documentation or readme)
- [ ] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [ ] 👌 Enhancement (improving an existing functionality like performance)
- [x] ✨ New feature (a non-breaking change that adds functionality)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📚 Description

- Created data-table-simple molecule for es-ds 3.0
- Created es-data-table-simple component based on es-ds 2.0
- Added data table simple link to ds-molecules-list.vue

### 🥼 Testing

- Tested the changes on http://localhost:8500/molecules/data-table-simple

#### 🧐 Feedback Requested / Focus Areas

- Whenever I add `table-responsive` class in es-data-table-simple the width does not exactly match the the width with [es-ds 2.0](https://design.energysage.dev/molecules/es-data-table-simple) designs. Whenever I dont have `table-responsive` class added it very closely matches the old designs so as of now I have not added the class to match the design. Does that work?
### 📝 Checklist

- [x] I have linked an issue or discussion.
- [ ] I have updated the documentation accordingly.
- [x] I have documented testing approach

### CODE REVIEW

File: es-ds-components/components/es-data-table-simple.vue [hunk 1]
```diff
@@ -0,0 +1,61 @@
+<template>
+  <div :class="tableClass">
```
Comment (nathanielwarner) [inline es-ds-components/components/es-data-table-simple.vue:2]: It seems like we should apply the `table-responsive` class here based on the [bootstrap docs](https://getbootstrap.com/docs/4.6/content/tables/#responsive-tables). I don't see any issue from that- were you seeing a problem with it put there?

File: es-ds-components/components/es-data-table-simple.vue [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+const isSingleCol = computed(() => !props.fields);
+const computedItems = computed(() => {
+  if (!props.items.length && !useSlots().default) {
+    throw new Error('Neither items prop nor slot content is defined.');
+  }
+  // If there are no fields, just return the items as a single-column array
+  if (!props.fields) {
+    return props.items;
+  }
```
Comment (nathanielwarner) [inline es-ds-components/components/es-data-table-simple.vue:57]: I notice in the old code there's these lines here:
```
if (this.fields.length !== this.items.length) {
    throw new Error('The number of fields and items must be the same.');
}
```
I'm wondering why you removed that check?

**Review by nathanielwarner — COMMENTED** [review]
Looks pretty good! Just a few comments inline

File: es-ds-components/components/es-data-table-simple.vue (hunk 1 above)
Comment (jaissica) [reply es-ds-components/components/es-data-table-simple.vue]: When I was applying `table-responsive` where we have the `table-borderless` class the width of each cell was affected, so I have now moved `table-responsive` to the the `div` tag and it works fine now. `table-borderless` works fine when applied to `div` tag or `table` class so I followed the [Bootstrap docs](https://getbootstrap.com/docs/4.6/content/tables/#borderless-table) and added it to `table` class.

Comment (nathanielwarner) [reply es-ds-components/components/es-data-table-simple.vue]: Nice!

**Review by nathanielwarner — APPROVED** [review]
LGTM! 🚀

**Pull request merged**